mesonbuild/backend/backends.py

```diff
@@ -1337,7 +1427,8 @@ def create_install_data(self) -> InstallData:
                         self.environment.get_build_dir(),
                         self.environment.get_prefix(),
                         strip_bin,
-                        self.environment.coredata.get_option(OptionKey('install_umask')),
+                        # This will only be a string or number, but mypy doesn't know that.
+                        T.cast(T.Union[str, int], self.environment.coredata.get_option(OptionKey('install_umask'))),
```

Comment:
Again, an assert will be more code, but we really want this to blow up if something goes wrong.